src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/Tool/MsAADTool.cs

```diff
@@ -61,22 +61,58 @@ public MsAADTool(string commandName, ProvisioningToolOptions provisioningToolOpt
             return null;
         }
 
+        private async Task<List<DirectoryObject>> GetGraphObjects()
+        {
+            List<DirectoryObject> graphObjectsList = new List<DirectoryObject>();
+            try
+            {
+                var graphObjects = await GraphServiceClient.Me.OwnedObjects
+                .Request()
+                .GetAsync();
+
+                if (graphObjects != null)
+                {
+                    graphObjectsList.AddRange(graphObjects.ToList());
+
+                    var nextPage = graphObjects.NextPageRequest;
+                    while (nextPage != null)
+                    {
+                        try
+                        {
+                            var additionalGraphObjects = await nextPage.GetAsync();
+                            if (additionalGraphObjects != null)
+                            {
+                                graphObjectsList.AddRange(additionalGraphObjects.ToList());
+                                nextPage = additionalGraphObjects.NextPageRequest;
+                            }
+                        }
+                        catch (ServiceException)
+                        {
+                            ConsoleLogger.LogMessage("Failed to retrieve all Azure AD/AD B2C objects(apps/service principals", LogMessageType.Error);
```

Comment:
nextPage = null;
Otherwise, you have an infinite loop here (I think)

Reply:
good catch fixed